include/boost/geometry/algorithms/centroid.hpp

```diff
@@ -352,15 +350,12 @@ struct centroid_multi
                 Point
             >::type state;
 
-        for (typename boost::range_iterator<Multi const>::type
-                it = boost::begin(multi);
-            it != boost::end(multi);
-            ++it)
+        for (auto it = boost::begin(multi); it != boost::end(multi); ++it)
```

Comment:
NOTE: we cannot do a range based for loop here
(as @awulkiew pointed out once).

In some places I changed it to a range based for loop, but these places are internal (for example turns).